src/librawspeed/decoders/DngDecoderSlices.cpp

```diff
@@ -418,10 +419,9 @@ void DngDecoderSlices::decodeSlice(DngDecoderThread* t) {
     while (!t->slices.empty()) {
       DngSliceElement e = t->slices.front();
       t->slices.pop();
-      LJpegPlain l(*mFile, e.byteOffset, e.byteCount, mRaw);
-      l.mDNGCompatible = mFixLjpeg;
+      DngDecompressor d(*mFile, e.byteOffset, e.byteCount, mRaw);
```

Comment:
Since the only aspect of my last comment you replied to was the question of whether or not to rename the LJpegDecompressor base class, it is not quite clear to me whether or not you are waiting for me to do that for you. As this has nothing to do with the purpose of this pr or the bug I fixed with it, I will not do that.